### PR TITLE
chore: dry run on dev to master merge pr

### LINKED ISSUE #2376: Dry run Play Release process for release PR
When creating a PR from development to master, the fastlane release command should be run, but in dry run mode. So, that issues like version code being old, changelog being too long, images being more, or description being long can be caught before merging into master

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #2376 

Changes: 
- updated script

### CODE REVIEW

File: scripts/update-apk.sh [hunk 1]
```diff
@@ -4,10 +4,7 @@ set -e
 git config --global user.name "Travis CI"
 git config --global user.email "noreply+travis@fossasia.org"
 
-export DEPLOY_BRANCH=${DEPLOY_BRANCH:-development}
-export PUBLISH_BRANCH=${PUBLISH_BRANCH:-master}
-
-if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_REPO_SLUG" != "fossasia/open-event-attendee-android" ] || ! [ "$TRAVIS_BRANCH" == "$DEPLOY_BRANCH" -o "$TRAVIS_BRANCH" == "$PUBLISH_BRANCH" ]; then
+if ! [[ ( $TRAVIS_PULL_REQUEST =~ ^(false)$ && $TRAVIS_REPO_SLUG =~ ^(fossasia/open-event-attendee-android)$ && $TRAVIS_BRANCH =~ ^(development|master)$ ) || ( $TRAVIS_BRANCH =~ ^(master)$ && $TRAVIS_PULL_REQUEST_BRANCH =~ ^(development)$ && $TRAVIS_REPO_SLUG =~ ^('fossasia/open-event-attendee-android')$ && $TRAVIS_PULL_REQUEST_SLUG =~ ^('fossasia/open-event-attendee-android')$ ) ]]; then
```
Comment (iamareebjamal) [inline scripts/update-apk.sh]: Too complex, I'm just a human, not a compiler, I need readable code

Define a variable. 

`export DRY_RUN = IS_PR_TO_MASTER`

Then OR it with the if condition

File: scripts/update-apk.sh [hunk 2]
```diff
@@ -45,6 +42,14 @@ for file in app*; do
     fi
 done
 
+if [[ $TRAVIS_BRANCH =~ ^(master)$ && $TRAVIS_PULL_REQUEST_BRANCH =~ ^(development)$ && $TRAVIS_REPO_SLUG =~ ^('fossasia/open-event-attendee-android')$ && $TRAVIS_PULL_REQUEST_SLUG =~ ^('fossasia/open-event-attendee-android')$ ]];then
```
Comment (iamareebjamal) [inline scripts/update-apk.sh]: Use the same dry run variable here

File: scripts/update-apk.sh [hunk 3]
```diff
@@ -4,10 +4,7 @@ set -e
 git config --global user.name "Travis CI"
 git config --global user.email "noreply+travis@fossasia.org"
 
-export DEPLOY_BRANCH=${DEPLOY_BRANCH:-development}
-export PUBLISH_BRANCH=${PUBLISH_BRANCH:-master}
-
-if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_REPO_SLUG" != "fossasia/open-event-attendee-android" ] || ! [ "$TRAVIS_BRANCH" == "$DEPLOY_BRANCH" -o "$TRAVIS_BRANCH" == "$PUBLISH_BRANCH" ]; then
```
Comment (iamareebjamal) [inline scripts/update-apk.sh]: Use the same dry run variable here

File: scripts/update-apk.sh (hunk 1 above)
Comment (adityastic) [reply scripts/update-apk.sh]: Oh yeah, better xD

File: scripts/update-apk.sh (hunk 3 above)
Comment (adityastic) [reply scripts/update-apk.sh]: Yes

File: scripts/update-apk.sh [hunk 4]
```diff
@@ -45,6 +42,14 @@ for file in app*; do
     fi
 done
 
+if [[ $TRAVIS_BRANCH =~ ^(master)$ && $TRAVIS_PULL_REQUEST_BRANCH =~ ^(development)$ && $TRAVIS_REPO_SLUG =~ ^('fossasia/open-event-attendee-android')$ && $TRAVIS_PULL_REQUEST_SLUG =~ ^('fossasia/open-event-attendee-android')$ ]];then
+    echo "cool, so you're trying to release a new version. Lets check things out first, hold on..."
+    cd ..
+    gem install fastlane
+    fastlane supply --aab ./apk/eventyay-attendee-master-app-playStore-release.aab --skip_upload_apk true --track alpha --json_key ./scripts/fastlane.json --package_name $PACKAGE_NAME --validate_only true
+    exit 0
+fi
```
Comment (iamareebjamal) [inline scripts/update-apk.sh]: Unnecessary duplicate code. 

Conditionally define a command

```sh
if DRY_RUN {
	export FASTLANE_DRY_RUN=--validate_only true
}
```

Then just append `FASTLANE_DRY_RUN` to the end of the fastlane command below `fastlane ... $PACKAGE_NAME $FASTLANE_DRY_RUN`

Comment (adityastic) [reply scripts/update-apk.sh]: got it

Comment (adityastic) [reply scripts/update-apk.sh]: @iamareebjamal ok so, If I dont duplicate this we will end up pushing to apk branch. currently we are terminating the script as soon as we validate, If I need to follow this I should seperate the sections inside the script into an ifelse condition. Let me know if you want me to go ahead with that approach

Comment (iamareebjamal) [reply scripts/update-apk.sh]: Move the fastlane block up and exit before branch pushing if it is dry run mode

**Comment by iamareebjamal** [comment]
Updates?

**Comment by adityastic** [comment]
> Updates?

Sorry, I was busy with class assessments from the last two days. Getting it done in a moment

**Comment by adityastic** [comment]
@iamareebjamal plz have a look

**Comment by iamareebjamal** [comment]
Didn't work 

> We upload apk only for changes in development or master, and not PRs. So, let's skip this shall we ? :)

**Comment by adityastic** [comment]
@iamareebjamal the pr needs to be from development branch. Not any branch

**Comment by iamareebjamal** [comment]
Should work from any branch

**Comment by adityastic** [comment]
@iamareebjamal oh ok. It was specified from development to master in the issue description. I'll change it tomorrow morning

**Comment by adityastic** [comment]
@iamareebjamal have a look

File: scripts/update-apk.sh [hunk 5]
```diff
@@ -17,34 +20,49 @@ fi
...
+if $IS_PUBLISH_BRANCH ;then
+    cd ..
+    gem install fastlane
+    if $PR_FOR_RELEASE ;then
+        echo "cool, so you're trying to release a new version. Lets check things out first, hold on..."
+        fastlane supply --aab ./apk/eventyay-attendee-master-app-playStore-release.aab --skip_upload_apk true --track alpha --json_key ./scripts/fastlane.json --package_name $PACKAGE_NAME $FASTLANE_DRY_RUN
+        exit 0
+    else
+        echo "Hurray....! Lets release a new build for our app"
+        fastlane supply --aab ./apk/eventyay-attendee-master-app-playStore-release.aab --skip_upload_apk true --track alpha --json_key ./scripts/fastlane.json --package_name $PACKAGE_NAME
```
Comment (iamareebjamal) [inline scripts/update-apk.sh]: Don't duplicate. That's why `$FASTLANE_DRY_RUN` should be assigned conditionally, not this statement

Comment (adityastic) [reply scripts/update-apk.sh]: @iamareebjamal have a look

**Comment by adityastic** [comment]
@iamareebjamal have a look

**Pull request merged**